cmd/argocd/commands/app.go

```diff
@@ -772,6 +779,41 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	return command
 }
 
+func printDiff(name string, live *unstructured.Unstructured, target *unstructured.Unstructured) {
+	tempDir, err := ioutil.TempDir("", "argocd-diff")
+	errors.CheckError(err)
+
+	targetFile := path.Join(tempDir, fmt.Sprintf("%s", name))
+	if target != nil {
+		data, err := yaml.Marshal(target)
+		errors.CheckError(err)
+		err = ioutil.WriteFile(targetFile, data, 0644)
+		errors.CheckError(err)
+	}
+
+	liveFile := path.Join(tempDir, fmt.Sprintf("%s-live.yaml", name))
+	if live != nil {
+		data, err := yaml.Marshal(live)
+		errors.CheckError(err)
+		err = ioutil.WriteFile(liveFile, data, 0644)
+		errors.CheckError(err)
+	}
+
+	cmdBinary := "diff"
+	var args []string
+	if envDiff := os.Getenv("KUBECTL_EXTERNAL_DIFF"); envDiff != "" {
+		parts, err := shlex.Split(envDiff)
+		errors.CheckError(err)
+		cmdBinary = parts[0]
+		args = parts[1:]
+	}
+
+	cmd := exec.Command(cmdBinary, append(args, liveFile, targetFile)...)
```

Comment:
If either live or target is nil, will the diff utility work since the file does not exist?